src/Hydra/Serializer/DocumentationNormalizer.php

```diff
@@ -396,13 +396,13 @@ private function getClasses(array $entrypointProperties, array $classes): array
                 [
                     '@type' => 'hydra:SupportedProperty',
                     'hydra:property' => [
-                        '@id' => '#ConstraintViolationList/violation',
-                        '@type' => 'rdf:Property',
-                        'rdfs:label' => 'violation',
+                        '@id' => '#ConstraintViolationList/violations',
+                        '@type' => 'hydra:Link',
```

Comment:
You can detect that it's an embedded relation because children documents contain an `@id` key.